server/routerlicious/packages/routerlicious-base/src/nexus/runner.ts

```diff
@@ -0,0 +1,222 @@
+		private readonly socketConnectTenantThrottler: IThrottler,
+		private readonly socketConnectClusterThrottler: IThrottler,
+		private readonly socketSubmitOpThrottler: IThrottler,
+		private readonly socketSubmitSignalThrottler: IThrottler,
+		private readonly storage: IDocumentStorage,
+		private readonly clientManager: IClientManager,
+		private readonly metricClientConfig: any,
+		private readonly throttleAndUsageStorageManager?: IThrottleAndUsageStorageManager,
+		private readonly verifyMaxMessageSize?: boolean,
+		private readonly redisCache?: ICache,
+		private readonly socketTracker?: IWebSocketTracker,
+		private readonly tokenRevocationManager?: ITokenRevocationManager,
+		private readonly revokedTokenChecker?: IRevokedTokenChecker,
+		private readonly collaborationSessionEventEmitter?: TypedEventEmitter<ICollaborationSessionEvents>,
+	) {}
+
+	// eslint-disable-next-line @typescript-eslint/promise-function-async
+	public start(): Promise<void> {
+		this.runningDeferred = new Deferred<void>();
+
+		// Create the HTTP server and attach nexus to it
+		const nexus = app.create(this.config, this.collaborationSessionEventEmitter);
+		nexus.set("port", this.port);
+
+		this.server = this.serverFactory.create(nexus);
+
+		const httpServer = this.server.httpServer;
+
+		const maxNumberOfClientsPerDocument = this.config.get(
+			"nexus:maxNumberOfClientsPerDocument",
+		);
+		const numberOfMessagesPerTrace = this.config.get("nexus:numberOfMessagesPerTrace");
+		const maxTokenLifetimeSec = this.config.get("auth:maxTokenLifetimeSec");
+		const isTokenExpiryEnabled = this.config.get("auth:enableTokenExpiration");
+		const isClientConnectivityCountingEnabled = this.config.get(
+			"usage:clientConnectivityCountingEnabled",
+		);
+		const isSignalUsageCountingEnabled = this.config.get("usage:signalUsageCountingEnabled");
+
+		httpServer.on("error", (error) => this.onError(error));
+		httpServer.on("listening", () => this.onListening());
+		httpServer.on("upgrade", (req, socket, initialMsgBuffer) => {
+			const metric = Lumberjack.newLumberMetric("WebSocket Connections", {
+				origin: "upgrade",
+				connections: socket.server._connections,
+			});
+			metric.success("WebSockets: connection upgraded");
+			socket.on("close", (hadError: boolean) => {
+				const closeMetric = Lumberjack.newLumberMetric("WebSocket Connections", {
+					origin: "close",
+					connections: socket.server._connections,
+					hadError: hadError.toString(),
+				});
+				closeMetric.success(
+					"WebSockets: connection closed",
+					hadError ? LogLevel.Error : LogLevel.Info,
+				);
+			});
+			socket.on("error", (error) => {
+				const errorMetric = Lumberjack.newLumberMetric("WebSocket Connections", {
+					origin: "error",
+					connections: socket.server._connections,
+					bytesRead: socket.bytesRead,
+					bytesWritten: socket.bytesWritten,
+					error: error.toString(),
+				});
+				errorMetric.success("WebSockets: connection error", LogLevel.Error);
```

Comment:
`success` on an `errorMetric` seems backwards... See my comment in reply to @sharmakhushboo's metric comment about unifying these 3 declarations of the same metric into 1.